cmd/backup/config.go

```diff
@@ -82,6 +83,16 @@ func (c *Config) resolveSecret(envVar string, secretPath string) (string, error)
 	return string(data), nil
 }
 
+func (c *Config) SetCompressionAlgo(algo string) error {
```

Comment:
If you wanted to model the limited number of options at config level (which I think is a good idea), you could define a custom `CompressionType` type and then create a custom decoder for the `BackupCompression` field. Something like this:

```go
type CompressionType string

func (c *CompressionType) Decode(v string) error {
	switch v {
	case "gz", "zst":
		*c = CompressionType(v)
		return nil
	default:
		return fmt.Errorf("config: error decoding compression type %s", v)
	}
}

func (c *CompressionType) String() string {
	return string(*c)
}

type Config struct {
	BackupCompression CompressionType `split_words:"true"`
}

```

For reference, this also already done for the `BackupExcludeRegexp` option.

Reply:
Nice, I like that